### PR TITLE
[te] Tiled (m=32 x n=32) gemm benchmark

### PR DESCRIPTION
**Pull request by bertmaher**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45906 [te][llvm] Enable fused multiply-add (fma) in code generation
* **#45905 [te] Tiled (m=32 x n=32) gemm benchmark**
* #45875 [te] Add a benchmark harness
* #45514 [te] Add a 2D convolution example test

Differential Revision: [D24142402](https://our.internmc.facebook.com/intern/diff/D24142402)

### CODE REVIEW

**Comment by bertmaher** [comment]
This schedule improves perf over the unscheduled version by 10x, and the code looks OK.  Tbh though, the perf improvement *should* be much higher (TVM gets 30x with the same schedule).  I think the issue is something subtle.  That said, to get the real prize here we need to put the inner block in registers (which also implies a better tile size like 6x16).

**Review by nickgg — APPROVED** [review]

File: benchmarks/cpp/tensorexpr/tensorexpr.cpp [hunk 1]
```diff
@@ -61,7 +61,72 @@ BENCHMARK_DEFINE_F(Gemm, TensorExprNoopt)(benchmark::State& state) {
...
+      },
+      {{K, "K"}});
+  te::LoopNest loop({CT});
+
+  {
+    auto const& loops = loop.getLoopStmtsFor(CT);
+    te::For* m = loops[0];
+    te::For* mo;
+    te::For* mi;
+    loop.splitWithMask(m, 32, &mo, &mi);
```
Comment (asuhan) [inline benchmarks/cpp/tensorexpr/tensorexpr.cpp:84]: Could you add a comment here (and below) why this split is needed?

**Review by asuhan — APPROVED** [review]

**Comment by nickgg** [comment]
@bertmaher for CPU perf, isn't SplitWithTail probably faster than SplitWithMask in the general case (although doesn't matter here because the tiling divides perfectly).

File: benchmarks/cpp/tensorexpr/tensorexpr.cpp (hunk 1 above)
Comment (bertmaher) [reply benchmarks/cpp/tensorexpr/tensorexpr.cpp:84]: Sure, what's the needed clarification?  Just like, why one would tile a matrix multiplication?

Comment (asuhan) [reply benchmarks/cpp/tensorexpr/tensorexpr.cpp:84]: I meant the mask version vs just split with tail, turns out I read too much into it per Nick's comment. I think it's ok as is.

**Comment by bertmaher** [comment]
I kinda suspect it depends on the actual dimensions.  I could see a strong case for splitWithTail on the M loop and splitWithMask on the N loop.  Basically because you're vectorizing over the N dimension so masked loads/stores are probably as fast as a tail loop, whereas M is broadcasted so you could actually skip work in a tail loop.

But yeah, I just wanted to do something that divides evenly for now.

**Pull request closed without merging**